### ISSUE TITLE
BadConstraints in slurm jobs

### ISSUE THREAD

**Issue by malinlarsson** [SciLifeLab/CAW#163]
I got a problem when running the version of the CAW that was available on January 23 (v0.9.9). 
The problem is that somehow the memory parameters in the slurm jobs is set to 131072:
#SBATCH --mem 131072

The memory is given in Mb which means that the above statement requests 131Gb, which is more than what is available on a normal node (128Gb). 
Also, it looks like I have this SLURM setting for all my jobs, also the ones running on just one core. 
I fetched the latest CAW code on January 23rd and just changed the runtimes to 94 hours, so I think this problem is in the latest CAW code somehow. 

I actually got an e-mail from Jessica Nettelblad at UPPMAX who had noticed that all my jobs were put on hold due to bad restrains, and she explained that it is best not to set the --mem flag at all. The memory is set automatically anyway to the maximum memory available on the number of cores we book. 

Here is what she wrote (translated to english by be):

"Remove the row with #SBATCH --mem 131072 from your slurm jobs, and the jobs will be started!

Some of the jobs hav now got stuck in the queue with error message BadCOnstraints. It is because of the requested amount of memory. It is not recommended to use the --mem parameter, because Uppmax automatically give access to a fraction of the memory available on the full node proportional to the number of cores booked for the job. If you book 3 cores you will get access to 3/16 of the total memory of the node. If you ask for 16 nodes you will get access to all of the memory for the node. 

Do you need more than 128Gb of memory? There are 17 nodes with 218 Gb and 17 nodes with 512 Gb. 
* To get at least 218 GB write -C fat 
* To get at least 512 GB write -C mem512GB.

So to summarise: Cancel the jobs that are marked with BadConstraints and start them without --mem. Hope it will work!"

**Comment by malinlarsson** [comment]
Here is the .nextflow.log and trace.txt files for CAW started January 23 on sample WNB111
[WNB111.trace.txt](https://github.com/SciLifeLab/CAW/files/726291/WNB111.trace.txt)

**Comment by malinlarsson** [comment]
new try:
[WNB111.nextflow.log.txt](https://github.com/SciLifeLab/CAW/files/726294/WNB111.nextflow.log.txt)

**Comment by malinlarsson** [comment]
Here is one relevant .command.run file from one of the mapping jobs:
[WNB111.command.run.txt](https://github.com/SciLifeLab/CAW/files/726311/WNB111.command.run.txt)

**Comment by malinlarsson** [comment]
I have checked, and it looks like we always use a fat node for jobs run on a full node. 
I have run 
jobstats -A b2015110 > jobstats_2015110.txt 

This captured all CAW runs after January 1 2017. 
To get info on process MapReads:
 grep 'nf-MapReads' jobstats_b2015110.txt | grep 'COMPLETED' | wc
gives 33 runs
grep 'nf-MapReads' jobstats_b2015110.txt | grep 'COMPLETED' | grep 'node_type_overbooked' | wc
gives 33 runs
The jobs have either flag
 node_type_overbooked:mem512GB:mem128GB
or
node_type_overbooked:mem256GB:mem128GB

Similar checks can be done for the other "full node" processes: 
grep 'nf-CreateIntervals' jobstats_b2015110.txt | grep 'COMPLETED' | wc
22
grep 'nf-CreateIntervals' jobstats_b2015110.txt | grep 'COMPLETED' | grep 'node_type_overbooked' | wc
18 (the jobs jobs  9493327, 9494708, 9496351 and 9498735 do not have data for some reason)

grep 'nf-ConcatVCF' jobstats_b2015110.txt | grep 'COMPLETED' | wc
     11     154    3422
grep 'nf-ConcatVCF' jobstats_b2015110.txt | grep 'COMPLETED' | grep 'node_type_overbooked' | wc
     10     140    3294
again, one job has no data (can't find it with finishedjobinfo jobid)

grep 'nf-RunStrelka' jobstats_b2015110.txt | grep 'COMPLETED' | wc
      5      70     994
grep 'nf-RunStrelka' jobstats_b2015110.txt | grep 'COMPLETED' | grep 'node_type_overbooked' | wc
      5      70     994

grep 'nf-RunManta' jobstats_b2015110.txt | grep 'COMPLETED' | wc
      5      70     984
 grep 'nf-RunManta' jobstats_b2015110.txt | grep 'COMPLETED' | grep 'node_type_overbooked' | wc
      5      70     984

**Comment by malinlarsson** [comment]
I have been trying to remove the demand for more than 128Gb memory by skipping the line 
 memory = 128.GB
under parameters in in my milou-slurm.config file. 
This creates SBATCH commands without the -mem flag, but the jobs are still run on fat nodes. 
The SBATCH  commands can be found in the .command.run file in the work folder of the process. 

**WITH memory=128**
Here is the beginning of a .command.run file for a CAW started WITH the memory=128 in milou-slurm.config. It corresponds to jobid 9648873
#!/bin/bash
#SBATCH -D /pica/v9/b2011185_nobackup/private/analysis/CAW/WNB112/work/4f/bc64c1c07485b9061af4e7e6448f24
#SBATCH -J nf-MapReads_(WNB112BT_12_2)
#SBATCH -o /pica/v9/b2011185_nobackup/private/analysis/CAW/WNB112/work/4f/bc64c1c07485b9061af4e7e6448f24/.command.log
#SBATCH -c 16
#SBATCH -t 48:00:00
#SBATCH --mem 131072
#SBATCH -p node
#SBATCH -A b2011185

Output from jobstats 9648873
9648873 milou COMPLETED malin   b2011185        'nf-MapReads_(WNB112BT_12_2)'   2017-01-24T00:58:57     04:58:53        node_type_overbooked:mem256GB:mem128GB  1620.1    16    m154    /sw/share/slurm/milou/uppmax_jobstats/m154/9648873

**Without memory=128:**
Here is the beginning of a .command.run file for a CAW started WITOUT the memory=128 in milou-slurm.config. It corresponds to jobid 9692902: 
#!/bin/bash
#SBATCH -D /pica/v9/b2011185_nobackup/private/analysis/CAW/WNB113/work/a5/67a7b8ad127ab302532460b5ce2fa7
#SBATCH -J nf-MapReads_(WNB113AT_1)
#SBATCH -o /pica/v9/b2011185_nobackup/private/analysis/CAW/WNB113/work/a5/67a7b8ad127ab302532460b5ce2fa7/.command.log
#SBATCH -c 16
#SBATCH -t 48:00:00
#SBATCH -p node
#SBATCH -A b2011185

Output from jobstats 9692902:
Running '/sw/uppmax/bin/finishedjobinfo -M milou -j 9692902' for more information, please be patient...
9692902 milou   COMPLETED       malin   b2011185        'nf-MapReads_(WNB113AT_1)'      2017-01-28T08:24:32     10:36:19        node_type_overbooked:mem256GB:mem128GB  16      20.4    16      m138    /sw/share/slurm/milou/uppmax_jobstats/m138/9692902

**Comment by szilvajuhos** [comment]
Solution is to set the memory to 110G.
